freezegun/api.py

```diff
@@ -371,10 +371,11 @@ def astimezone(self, tz=None):
     @classmethod
     def fromtimestamp(cls, t, tz=None):
         if tz is None:
-            return real_datetime.fromtimestamp(
-                    t, tz=dateutil.tz.tzoffset("freezegun", cls._tz_offset())
-                ).replace(tzinfo=None)
-        return datetime_to_fakedatetime(real_datetime.fromtimestamp(t, tz))
+            tz = dateutil.tz.tzoffset("freezegun", cls._tz_offset())
+            result = real_datetime.fromtimestamp(t, tz=tz).replace(tzinfo=None)
+        else:
+            result = datetime_to_fakedatetime(real_datetime.fromtimestamp(t, tz))
```

Comment:
```suggestion
            result = real_datetime.fromtimestamp(t, tz)
```

The `datetime_to_fakedatetime` already happens at the end - no need to duplicate this call.

Happy to merge once that's fixed.

Reply:
fixed. thanks for pointing this out.